gandalf.js

```diff
@@ -8,71 +8,78 @@
  *
```

Comment:
SCRIPT-This logic might be clearer if you used a switch statement instead of multiple if-else statements.